server/src/models/Pokemon.js

```diff
@@ -415,22 +519,37 @@ module.exports = class Pokemon extends Model {
     return finalResults
   }
 
-  static async evalQuery(mem, query) {
-    return (
-      (mem
-        ? fetchJson(mem, {
-            method: 'POST',
-            headers: {
-              Accept: 'application/json',
-              'Content-Type': 'application/json',
-            },
-            body: query.toKnexQuery().toString(),
-          })
-        : query) || []
-    )
+  static async evalQuery(mem, query, method = 'POST') {
+    if (queryDebug) {
+      if (!fs.existsSync(resolve(__dirname, './queries'))) {
+        fs.mkdirSync(resolve(__dirname, './queries'), { recursive: true })
+      }
+      if (mem && typeof query === 'string') {
+        fs.writeFileSync(
+          resolve(__dirname, './queries', `${Date.now()}.json`),
+          query,
+        )
+      } else {
+        fs.writeFileSync(
+          resolve(__dirname, './queries', `${Date.now()}.txt`),
+          query.toKnexQuery().toString(),
```

Comment:
When querying available, the query is undefined, so you might want to check that we at least have a query.